pkg/util/common/common.go

```diff
@@ -39,6 +39,15 @@ func (s StringSet) GetAll() []string {
 // StructToMap converts a struct to a map[string]interface{} based on `json` annotations defaulting to field names
 func StructToMap(obj interface{}) map[string]interface{} {
 	rt, rv := reflect.TypeOf(obj), reflect.ValueOf(obj)
+
+	if rt == nil {
+		return nil
+	}
```

Comment:
made a fairly simple addition here if a nil interface or error is passed in.  Returning nil will actually return map[string]interface{}(nil) in line with the specified return type.  Let me know if there's a better way to handle this, debated adding an error return type but then downstream functions will have to break out the returns.

Reply:
Could also return an empty map. @vboulineau have an opinion?

Reply:
I meant return `map[string]interface{}{}` for the `if rt == nil` case.

Reply:
done (changed to `map[string]interface{}{} `for the `nil` case)